src/hooks/useConnectedDelegate.ts

```diff
@@ -24,35 +30,54 @@ const useConnectedDelegate = () => {
     enabled: !!address,
     queryKey: ["useConnectedDelegate", address, refetchDelegate, retries],
     queryFn: async () => {
-      const [delegate, advancedDelegators, balance] =
-        await fetchConnectedDelegate(address!);
+      const [delegate, advancedDelegators, balance] = await Promise.all([
+        api.get(`/delegates/${address}`),
+        api.get(`/delegates/${address}/delegation-chains`),
+        contracts.token.contract.balanceOf(address as `0x${string}`),
+      ]);
       if (refetchDelegate) {
         revalidateDelegateAddressPage(refetchDelegate.address);
       }
       setLastVotingPower(delegate.votingPower);
 
       // If refetchDelegate?.votingPower we are looking for a revalidation on the page of the delegatee
       if (refetchDelegate?.prevVotingPowerDelegatee) {
-        const delegatee = await fetchDelegate(refetchDelegate.address);
+        const delegatee = await api.get(
+          `/delegates/${refetchDelegate.address}`
+        );
         /**
          * Materialized view that brings the new voting power takes one minute to sync
          * Refetch delegate will be set to null by the delegateProfileImage
          */
         if (
           delegatee.votingPower === refetchDelegate.prevVotingPowerDelegatee
         ) {
-          await timeout(2000);
-          const _retries = retries + 1;
-          setRetries(_retries);
+          // Check if maximum retries has been reached
+          if (retries < MAX_RETRIES) {
+            // Implement exponential backoff
```

Comment:
@ferrodri nice work on the retries. can we refac this into something reusable for other components? we'll probably need safe retry logic in other places in our app

Reply:
will implement this in another PR